omod/src/main/java/org/openmrs/module/metrics/web/controller/DefaultMetricsServlet.java

```diff
@@ -0,0 +1,74 @@
+package org.openmrs.module.metrics.web.controller;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.time.LocalDateTime;
+import java.util.SortedMap;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.openmrs.module.metrics.api.exceptions.MetricsException;
+import org.openmrs.module.metrics.util.MetricHandler;
+
+public class DefaultMetricsServlet extends HttpServlet {
+
+	private transient MetricRegistry metricRegistry;
+
+	private transient ObjectMapper objMapper;
+
+	private LocalDateTime startDatetime;
+
+	private LocalDateTime endDatetime;
```

Comment:
None of these should be class level variables. They exist for the lifetime of a single request and so should exist inside the `doGet()` method.

Reply:
sure